modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java

```diff
@@ -63,342 +54,273 @@
+            taskDescription.setName(TASK_PREFIX + name + i);
+            taskDescription.setTaskGroup(MessageProcessorConstants.SCHEDULED_MESSAGE_PROCESSOR_GROUP);
+            /*
+             * If this interval value is less than 1000 ms, ntask will throw an
+             * exception while building the task. So to get around that we are
+             * setting threshold interval value of 1000 ms to the task
+             * description here. But actual interval value may be less than 1000
+             * ms, and hence isThrotling is set to TRUE.
+             */
+            if (interval < MessageProcessorConstants.THRESHOULD_INTERVAL) {
+                taskDescription.setInterval(MessageProcessorConstants.THRESHOULD_INTERVAL);
+            } else {
+                taskDescription.setInterval(interval);
             }
-        } catch (SchedulerException e) {
-            throw new SynapseException("Error starting the scheduler", e);
-        }
-
-        Trigger trigger;
-        TriggerBuilder<Trigger> triggerBuilder = newTrigger().withIdentity(name + "-trigger");
-
-        if (cronExpression == null || "".equals(cronExpression)) {
-            trigger = triggerBuilder
-                    .withSchedule(simpleSchedule()
-                        .withIntervalInMilliseconds(isThrottling(this.interval) ? 1000 : this.interval)
-                        .repeatForever()
-                        .withMisfireHandlingInstructionNextWithRemainingCount())
-                    .build();
-        } else {
-            trigger = triggerBuilder
-                    .startNow()
-                    .withSchedule(CronScheduleBuilder.cronSchedule(cronExpression)
-                        .withMisfireHandlingInstructionDoNothing())
-                    .build();
-        }
-
-        JobDataMap jobDataMap = getJobDataMap();
-        jobDataMap.put(MessageProcessorConstants.PARAMETERS, parameters);
-
-        JobBuilder jobBuilder = getJobBuilder();
-        JobDetail jobDetail = jobBuilder.usingJobData(jobDataMap).build();
-
-        try {
-            scheduler.scheduleJob(jobDetail, trigger);
-        } catch (SchedulerException e) {
-            throw new SynapseException("Error scheduling job : " + jobDetail
-                    + " with trigger " + trigger, e);
+            taskDescription.setIntervalInMs(true);
+            taskDescription.addResource(TaskDescription.INSTANCE, task);
+            taskDescription.addResource(TaskDescription.CLASSNAME, task.getClass().getName());
+
+            /*
+             * If there is a Cron Expression we need to set it into the
+             * TaskDescription so that the framework will take care of it.
+             */
+            if (cronExpression != null) {
+                taskDescription.setCronExpression(cronExpression);
+            }
+            taskManager.schedule(taskDescription);
         }
+        logger.info("Started message processor. [" + getName() + "].");
```

Comment:
These issues were fixed in pull request https://github.com/wso2/wso2-synapse/pull/242